### PR TITLE
[Merged by Bors] - feat: port logic.equiv.defs

### PR DESCRIPTION
**Pull request by kim-em**
mathlib3 sha: 7951ed37deb3b2923a6f47d9bdcb4d69a8703550

~~Waiting on the release of [nightly-2022-11-17](https://github.com/leanprover/lean4-nightly/releases/tag/nightly-2022-11-17).~~

- [x] depends on: #541
- [x] depends on: #559

### CODE REVIEW

**Comment by Vierkantor** [comment]
The last parts of this file depend on `data.quot` via `logic.equiv.defs → tactic.basic → tactic.trunc_cases → data.quot`. Otherwise everything builds with not too many hacks or backports. But see also: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60simp.60.20failing.20on.20partial.20application.20of.20projections

File: Mathlib/Init/Data/Bool/Lemmas.lean [hunk 1]
```diff
@@ -123,10 +123,15 @@ theorem coe_sort_false : (↥false : Prop) = False := by simp
...
+@[simp] theorem decide_eq_true (b : Bool) {h} : @decide (b = true) h = b :=
+by cases b
+   . simp
+   . simp
+
 theorem decide_iff (p : Prop) [d : Decidable p] : decide p = true ↔ p := by simp
 #align to_bool_iff Bool.decide_iff
 
-theorem decide_true {p : Prop} [Decidable p] : p → decide p :=
+theorem decide_true (p : Prop) [Decidable p] : p → decide p :=
```
Comment (Vierkantor) [inline Mathlib/Init/Data/Bool/Lemmas.lean:134]: Not sure how I changed this...
```suggestion
theorem decide_true {p : Prop} [Decidable p] : p → decide p :=
```

**Comment by semorrison** [comment]
I'm so confused. This builds locally for me.

**Comment by semorrison** [comment]
This is failing on the issue reported at https://github.com/leanprover/lean4/issues/1822.

**Comment by semorrison** [comment]
And now failing on the issue reported at https://github.com/leanprover/lean4/issues/1829

**Comment by Ruben-VandeVelde** [comment]
I merged the nightly-2022-11-17 bump, hope that was alright

**Pull request closed without merging**